basculechecks/capabilitiesvalidator.go

```diff
@@ -130,25 +144,33 @@ func (c CapabilitiesValidator) checkCapabilities(capabilities []string, reqURL s
 
 // getCapabilities runs some error checks while getting the list of
 // capabilities from the attributes.
-func getCapabilities(attributes bascule.Attributes) ([]string, string, error) {
+func getCapabilities(attributes bascule.Attributes) ([]string, error) {
 	if attributes == nil {
-		return []string{}, UndeterminedCapabilities, ErrNilAttributes
+		return []string{}, ErrNilAttributes
 	}
 
 	val, ok := attributes.Get(CapabilityKey)
 	if !ok {
-		return []string{}, UndeterminedCapabilities, fmt.Errorf("couldn't get capabilities using key %v", CapabilityKey)
+		err := errWithReason{
+			err:    fmt.Errorf("couldn't get capabilities using key %v", CapabilityKey),
+			reason: UndeterminedCapabilities,
+		}
+		return []string{}, err
 	}
 
 	vals, err := cast.ToStringSliceE(val)
 	if err != nil {
-		return []string{}, UndeterminedCapabilities, fmt.Errorf("capabilities \"%v\" not the expected string slice: %v", val, err)
+		err = errWithReason{
```

Comment:
If applicable, you could have an exported sentinel errWithReason (i.e. ErrNilAttributes) and wrap it with these error details.

Reply:
Same advice for some of those errors in basculechecks/metricvalidator.go